src/Skia/Avalonia.Skia/DrawingContextImpl.cs

```diff
@@ -1158,7 +1158,10 @@ private void ConfigureTileBrush(ref PaintWrapper paintWrapper, Rect targetBox, I
 
             if (content.Transform is not null)
             {
-                transform = content.Transform.Value * transform;
+                var transformOrigin = content.TransformOrigin.ToPixels(targetRect);
+                var offset = Matrix.CreateTranslation(transformOrigin);
+
+                transform *= -offset * content.Transform.Value * offset;
```

Comment:
Does this actually work? I think `-offset` also flips X and Y coordinates, but the test images won't catch this because they are symmetric in X and Y.

Reply:
TransformOrigin mainly used to move the origin to some rotation point

The math is correct here

Reply:
Sorry, only noticed now that `-offset` actually means the inverse of `offset`. I think it would be more efficient to `CreateTranslation(-transformOrigin)`, but performance is probably not a problem here.